ExampleGallery/UAP/CameraEffectExample.xaml.cs

```diff
@@ -137,20 +155,28 @@ private void EffectCombo_SelectionChanged(object sender, SelectionChangedEventAr
         {
             await ResetEffectsAsync();
 
-            string typeName = null;
+            VideoEffectDefinition videoEffectDefinition = null;
 
             switch (effect)
             {
-                case displacementEffect: typeName = typeof(DisplacementEffect).FullName; break;
-                case rotatingTilesEffect: typeName = typeof(RotatedTilesEffect).FullName; break;
+                case displacementEffect:
+                    videoEffectDefinition = new VideoEffectDefinition(typeof(DisplacementEffect).FullName, new PropertySet());
+                    break;
+                case rotatingTilesEffect:
+                    videoEffectDefinition = new VideoEffectDefinition(typeof(RotatedTilesEffect).FullName, new PropertySet());
+                    break;
+                case variableEffect:
+                    variableEffectValueSlider.Visibility = Visibility.Visible;
+                    effectPropertySet = new PropertySet();
```

Comment:
effectPropertySet is created here, but never set back to null when switching to different effects.

Suggestion: if you initialize effectPropertySet at class scope so it is always valid, you would no longer need the checks for it being null that occur in other places.  Then you can always pass that property set into the VideoEffectDefinition constructor instead of calling new PropertySet (regardless of the type of effect) so this init code would not need to change so much.  Leave it using typeName, and just include extra code to initialize the property value when creating your new effect type without having to change init code for other effects at all.